src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -186,7 +194,19 @@ const connectInfiniteHits: InfiniteHitsConnector = (
         // We're doing this to "reset" the widget if a refinement or the
         // query changes between renders, but we want to keep it as is
         // if we only change pages.
-        const { page = 0, ...currentState } = state;
+        const {
+          page = 0,
+          hierarchicalFacets,
+          disjunctiveFacets,
+          ...currentState
+        } = state;
+
+        currentState.hierarchicalFacetsRefinements = filterEmptyRefinements(
+          currentState.hierarchicalFacetsRefinements
+        );
+        currentState.disjunctiveFacetsRefinements = filterEmptyRefinements(
```

Comment:
how come this isn't necessary for numeric?

Reply:
Oh, it's necessary for numeric too, just forgot to add it. Fixed in https://github.com/algolia/instantsearch.js/pull/4104/commits/31d7cdba144f3fd9c3205382d6808604ba49727a